cpp/oneapi/dal/algo/objective_function/backend/gpu/compute_kernel_dense_batch_impl_dpc.cpp

```diff
@@ -0,0 +1,272 @@
+                                                     out_gradient,
+                                                     Float(0),
+                                                     Float(0),
+                                                     { prob_e, fill_event });
+                grad_event.wait_and_throw();
+                const auto* const grad_ptr = out_gradient.get_data();
+                auto update_event = q_.submit([&](sycl::handler& cgh) {
+                    cgh.depends_on({ grad_event });
+
+                    const auto range = oneapi::dal::backend::make_range_1d(p + 1);
+
+                    cgh.parallel_for(range, [=](sycl::id<1> idx) {
+                        ans_grad_ptr[idx] += grad_ptr[idx];
+                    });
+                });
+                update_event.wait_and_throw();
+            }
+        }
+
+        if (desc.get_result_options().test(result_options::hessian)) {
+            auto fill_event = fill(q_, out_hessian, Float(0), {});
+            auto hess_event = compute_hessian(q_,
+                                              params_nd,
+                                              data_nd,
+                                              responses_nd,
+                                              probabilities,
+                                              out_hessian,
+                                              Float(0),
+                                              Float(0),
+                                              { prob_e, fill_event });
+            hess_event.wait_and_throw();
+            auto* const ans_hess_ptr = ans_hessian.get_mutable_data();
+            const auto* const hess_ptr = out_hessian.get_data();
+            auto update_event = q_.submit([&](sycl::handler& cgh) {
+                cgh.depends_on({ hess_event });
+                const auto range = oneapi::dal::backend::make_range_2d(p + 1, p + 1);
+                cgh.parallel_for(range, [=](sycl::id<2> idx) {
+                    ans_hess_ptr[idx[0] * (p + 1) + idx[1]] += hess_ptr[idx[0] * (p + 1) + idx[1]];
+                });
+            });
+            update_event.wait_and_throw();
+        }
+    }
+    if (desc.get_result_options().test(result_options::value)) {
+        auto update_event = q_.submit([&](sycl::handler& cgh) {
+            cgh.single_task([=] {
+                ans_loss_ptr[0] /= n;
+                for (std::int64_t i = 1; i < p + 1; ++i) {
+                    ans_loss_ptr[0] +=
+                        L2 * params_ptr[i] * params_ptr[i] + L1 * sycl::abs(params_ptr[i]);
+                }
+            });
+        });
+        update_event.wait_and_throw();
+    }
+    if (desc.get_result_options().test(result_options::gradient)) {
+        auto update_event = q_.submit([&](sycl::handler& cgh) {
+            const auto range = oneapi::dal::backend::make_range_1d(p + 1);
+            cgh.parallel_for(range, [=](sycl::id<1> idx) {
+                ans_grad_ptr[idx[0]] /= n;
```

Comment:
`n` value is common for all work items. Please move it out

1. Create a variable outside of the kernel. Let say `const Float inv_n = Float(1) / n;`
2. Use it inside of the kernel